### PR TITLE
docs: PDE Tutorial

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
This PR introduces a tutorial for Palette Dev Engine featuring how to deploy applications into a virtual cluster.
💻 [Preview](https://deploy-preview-1091--docs-spectrocloud.netlify.app/devx/apps/deploy-app)
🎫 [DOC-421](https://spectrocloud.atlassian.net/browse/DOC-421)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1091#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 1]
```diff
@@ -0,0 +1,1180 @@
...
+```shell
+terraform apply -auto-approve
+```
+
+```
+// Output condensed for readability
+Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
+```
+
+Log in to [Palette](https://console.spectrocloud.com) and navigate to the left **Main Menu**, select **Apps**. Click on the **scenario-1** row, which will take you to the application’s overview page. Once you are on the scenario-1 overview page, click on the exposed URL for the service. A hyperlink for port 8080 is available.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:773]:
```suggestion
Log in to [Palette](https://console.spectrocloud.com), navigate to the left **Main Menu**, and select **Apps**. Click on the **scenario-1** row, which takes you to the application’s overview page. Once you are on the scenario-1 overview page, click on the exposed URL for the service. A hyperlink for port 8080 is available.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 2]
```diff
@@ -0,0 +1,1180 @@
...
+
+To deploy the second scenario, you will again deploy the same three resource types previously discussed but another instance of them.
+
+- `spectrocloud_virtual_cluster` - `cluster-2` - this resource will create the second virtual cluster. 
+
+
+- `spectrocloud_application_profile` - `hello-universe-complete` - the application profile that will contain the three different services, database, API, and UI
+
+
+- `spectrocloud_application` - `scenario-2` - the application that will be deployed into cluster-2 that use the `spectrocloud_application_profile.hello-universe-complete` app profile.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:812]:
```suggestion
- `spectrocloud_application` - `scenario-2` - the application that will be deployed into cluster-2 that uses the `spectrocloud_application_profile.hello-universe-complete` app profile.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 3]
```diff
@@ -0,0 +1,1180 @@
...
+            serviceType: LoadBalancer
+    EOT   
+  }
+  tags = concat(var.tags, ["scenario-2"])
+}
+```
+
+Each service has its own `pack {}` and a set of unique properties and values. 
+
+The database service block uses a different data resource, `data.spectrocloud_pack_simple.postgres_service`,  to find the Postgres service. If you review the data resource, you will find a different type, `operator-instance`. The Postgres service uses a Postgres [operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) to manage the database inside the virtual cluster.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:977]:
```suggestion
The database service block uses a different data resource, `data.spectrocloud_pack_simple.postgres_service`, to find the Postgres service. If you review the data resource, you will find a different type, `operator-instance`. The Postgres service uses a Postgres [operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) to manage the database inside the virtual cluster. 
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 4]
```diff
@@ -0,0 +1,1180 @@
...
+
+<InfoBox>
+
+All container services expose their service address, Kubernetes hostname, and the exposed service ports as output variables.
+You will use output variables frequently when creating app profiles in the future. You can learn more about connecting services by refering to the [Service Connectivity](/devx/app-profile/services/connectivity) documentation.  
+
+</InfoBox>
+
+
+Open up the file **inputs.tf** and change the `variable enable-second-scenario"` and set its default value to `true`.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1093]:
```suggestion
Open the **inputs.tf** file and set the `variable enable-second-scenario"` default value to `true`.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 5]
```diff
@@ -0,0 +1,1180 @@
...
+
+```terraform
+variable "enable-second-scenario" {
+  type        = bool
+  description = "Whether to enable the second scenario"
+  default     = true
+}
+```
+
+Next, issue the command `terraform apply` to deploy the second scenario. Take notice how the `-var` flag is included with the token value in the command.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1105]:
```suggestion
Next, issue the command `terraform apply` to deploy the second scenario. Notice how the `-var` flag is included with the token value in the command. 
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 6]
```diff
@@ -0,0 +1,1180 @@
...
+</Tabs.TabPane>
+
+</Tabs>
+
+
+# Wrap-Up
+
+In this tutorial, you learned about Palette’s Dev Engine and App Mode. You deployed two virtual clusters, each containing a different architecture and configuration of the Hello Universe application. Palette’s Dev Engine enables developers to quickly deploy applications into a Kubernetes environment without requiring Kubernetes knowledge. In a matter of minutes, you deployed a new Kubernetes cluster and all its applications without having to write Kubernetes configuration files.
+
+To learn more about Palette Dev Engine and its capabilities check out the reference resource below.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1172]:
```suggestion
To learn more about Palette Dev Engine and its capabilities, check out the reference resource below.
```

**Review by ritawatson — DISMISSED** [review]
A clear tutorial.  A couple suggestions:
- Remove 'on' in 'click on': Change 'click on _element_' to 'click _element_'. I suggest this as part of our internal style. 
- While 'Go ahead and ...' is intended to keep the tone friendly, it seems a bit overused in this tutorial and it adds unnecessary verbiage. Maybe keep it in a hand full of places but remove when introducing a list.

Do forward slashes need to be added in image paths now that images are in one folder?

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1091#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1091#pullrequestreview-1296561603" title="Approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 7]
```diff
@@ -0,0 +1,1180 @@
...
+This time you will use a different container image for the UI that contains a reverse proxy. The reverse proxy is responsible for two important tasks. The first task is to route requests to the API inside the UI container instead of using your browser. The reason behind this behavior is to access the API server that is not publicly exposed. The reverse proxy will pick up the API request and forward it to the API container. The second task is to insert an authentication token in all requests to the API. The API server configuration you provided enabled authentication, so a Bearer token is required for all requests. The following diagram illustrates the network connectivity path and behavior discussed.
+
+  
+  
+
+![A diagram of the reverse proxy architecture](/tutorials/deploy-app/devx_apps_deploys-apps_reverse-proxy-diagram.png)
+
+  
+
+Go ahead and provide the UI container with the following information.
```
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:317]:
```suggestion
Provide the UI container with the following information.
```

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1091#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 8]
```diff
@@ -0,0 +1,1255 @@
...
+      value: "true"
+    - name: DB_ENCRYPTION
+      value: "${var.database-ssl-mode}"
+    - name: AUTHORIZATION
+      value: "true"
+    EOT
+  }
+```
+
+The last `pack {}` block in the app profile resource `spectrocloud_application_profile.hello-universe-complete` is for the UI. Like the API service, environment variables are used to initialize UI and the reverse proxy. The UI service requires the URL of the API service and the URL of the public-facing load balancer. Palette output variables are used to populate these two environment variables. A Terraform variable will populate the authentication token required for all API requests.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1127]:
```suggestion
The last `pack {}` block in the app profile resource `spectrocloud_application_profile.hello-universe-complete` is for the UI. Like the API service, environment variables are used to initialize the UI and the reverse proxy. The UI service requires the URL of the API service and the URL of the public-facing load balancer. Palette output variables are used to populate these two environment variables. A Terraform variable will populate the authentication token required for all API requests.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 9]
```diff
@@ -0,0 +1,1255 @@
...
+```shell
+terraform apply -var="token=931A3B02-8DCC-543F-A1B2-69423D1A0B94" -auto-approve
+```
+
+```
+// Output condensed for readability
+Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
+```
+
+Log in to [Palette](https://console.spectrocloud.com) and navigate to the left **Main Menu**, click on **Apps**. Select the **scenario-2** row. Once you are on the scenario-2 overview page, click on the exposed URL for the service. A hyperlink for port 8080 and port 3000 is available.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1193]:
```suggestion
Log in to [Palette](https://console.spectrocloud.com) and navigate to the left **Main Menu**, click on **Apps**. Select the **scenario-2** row. When you are on the scenario-2 overview page, click on the exposed URL for the service. A hyperlink for port 8080 and port 3000 is available.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 10]
```diff
@@ -0,0 +1,1255 @@
...
+<WarningBox>
+
+  
+It takes between one to three minutes for DNS to properly resolve the public load balancer URL. We recommend waiting a few moments before clicking on the service URL to prevent the browser from caching an unresolved DNS request.
+
+</WarningBox>
+
+![View of the self-hosted hello universe app](/tutorials/deploy-app/devx_apps_deploy-app_self-hosted-hello-universe.png)
+
+The global counter is no longer available; instead, you have a counter that starts at zero. Each time you click on the center image, the counter is incremented and stored in the Postgres database along with metadata.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1210]:
```suggestion
The global counter is no longer available. Instead, you have a counter that starts at zero. Each time you click on the center image, the counter is incremented and stored in the Postgres database along with metadata.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 11]
```diff
@@ -0,0 +1,1255 @@
...
+
+
+## Clean-up
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1213]:
```suggestion
## Cleanup
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 12]
```diff
@@ -0,0 +1,1255 @@
...
+```
+Destroy complete! Resources: 6 destroyed.
+```
+
+<br />
+
+<InfoBox>
+
+
+If a cluster remains in the delete phase for over 15 minutes, it becomes eligible for **Force Delete**. To trigger a force delete, navigate to the respective cluster’s details page and click on **Settings**. Click on the **Force Delete Cluster** to delete the cluster. Palette will automatically remove clusters stuck in the cluster deletion phase for over 24 hrs.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1232]:
```suggestion
If a cluster remains in the delete phase for over 15 minutes, it becomes eligible for **Force Delete**. To trigger a force delete, navigate to the respective cluster’s details page and click on **Settings**. Click on the **Force Delete Cluster** to delete the cluster. Palette will automatically remove clusters stuck in the cluster deletion phase for over 24 hours.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 13]
```diff
@@ -0,0 +1,1255 @@
...
+</Tabs.TabPane>
+
+</Tabs>
+
+
+# Wrap-Up
+
+In this tutorial, you learned about Palette’s Dev Engine and App Mode. You deployed two virtual clusters, each containing a different architecture and configuration of the Hello Universe application. Palette’s Dev Engine enables developers to quickly deploy applications into a Kubernetes environment without requiring Kubernetes knowledge. In a matter of minutes, you deployed a new Kubernetes cluster and all its applications without having to write Kubernetes configuration files.
+
+To learn more about Palette Dev Engine and its capabilities, check out the reference resource below.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1247]:
```suggestion
To learn more about Palette Dev Engine and its capabilities, check out the reference resources below.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 14]
```diff
@@ -0,0 +1,1255 @@
...
+  
+
+![Delete a cluster view with arrow](/tutorials/deploy-app/devx_apps_deploy-apps_delete-cluster-view.png)
+
+  
+
+<InfoBox>
+ 
+
+If a cluster remains in the delete phase for over 15 minutes, it becomes eligible for **Force Delete**. To trigger a force delete, navigate to the respective cluster’s details page and click on **Settings**. Click on the **Force Delete Cluster** to delete the cluster. Palette will automatically remove clusters stuck in the cluster deletion phase for over 24 hrs.
```
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:432]:
```suggestion
If a cluster remains in the delete phase for over 15 minutes, it becomes eligible for **Force Delete**. To trigger a force delete, navigate to the respective cluster’s details page and click on **Settings**. Click on the **Force Delete Cluster** to delete the cluster. Palette will automatically remove clusters stuck in the cluster deletion phase for over 24 hours.
```

**Review by ritawatson — APPROVED** [review]
Nice tone and clear instruction. Just some some fit and finish remarks.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1091#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1091#pullrequestreview-1320903404" title="Approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by tiwarisumit3 — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1091#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1091#pullrequestreview-1320903404" title="Approved">ritawatson</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1091#pullrequestreview-1321046204" title="Approved">tiwarisumit3</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 15]
```diff
@@ -0,0 +1,1255 @@
...
+
+
+There are no expenses associated with this tutorial as everything falls under the Palette Free Tier.
+
+
+# Architecture 
+
+The tutorial includes two scenarios, and for each scenario, you will deploy a separate Kubernetes environment. The following diagram illustrates the different layers that will power the tutorial environment.
+
+![Architecutre diagram depicting two virtual clusters](/tutorials/deploy-app/devx_apps_deploy-apps_architecture-diagram.png)
```
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:42]:
```suggestion
![Architecture diagram depicting two virtual clusters](/tutorials/deploy-app/devx_apps_deploy-apps_architecture-diagram.png)
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 16]
```diff
@@ -0,0 +1,1255 @@
...
+<br />
+
+```tf
+data "spectrocloud_cluster_group" "beehive" {
+  name    = var.cluster-group-name
+  context = "system"
+}
+```
+
+Next, take a look at the the **application-profiles.tf** file. The resource `spectrocloud_application_profile.hello-universe-ui` is the resource responsible for creating the app profile for the first scenario. There are several points of interest in this resource that you should be familiar with. Focus on these five key points:
```
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:590]:
```suggestion
Next, take a look at the **application-profiles.tf** file. The resource `spectrocloud_application_profile.hello-universe-ui` is the resource responsible for creating the app profile for the first scenario. There are several points of interest in this resource that you should be familiar with. Focus on these five key points:
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md [hunk 17]
```diff
@@ -0,0 +1,1255 @@
...
+</PointsOfInterest>
+
+
+A tip for gathering the required values to provide the `values` attribute is to visit the Palette console and create the app profile through the UI. During the app profile creation process, click on the API button to display the API payload. Review the payload's `values` attribute to find all of the properties of the service. You can copy the entire string and pass it to the resource `spectrocloud_application_profile` as an input for the `values` attribute.
+
+
+![UI's ability to display the API object](/tutorials/deploy-app/devx_apps_deploy-apps_ui-api-display.png)
+
+
+The last Terraform resource to review before deploying the application is located in the **application.tf** file. The resource `spectrocloud_application.hello-universe-ui` is what creates the *App*. In Palette, an app combines a virtual cluster and an app profile. When you deploy an app profile into a virtual cluster, you create an app. This resource points to the app profile `spectrocloud_application_profile.hello-universe-ui` and the cluster resource `spectrocloud_virtual_cluster.cluster-1`. The two resources are required to create an app.
```
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:739]:
```suggestion
The last Terraform resource to review before deploying the application is located in the **application.tf** file. The resource `spectrocloud_application.hello-universe-ui` is what creates the *app*. In Palette, an app combines a virtual cluster and an app profile. When you deploy an app profile into a virtual cluster, you create an app. This resource points to the app profile `spectrocloud_application_profile.hello-universe-ui` and the cluster resource `spectrocloud_virtual_cluster.cluster-1`. The two resources are required to create an app.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md (hunk 11 above)
Comment (karl-cardenas-coding) [reply content/docs/04.5-devx/06-apps/05-deploy-app.md:1213]: @ritawatson We have "clean-up" in the tutorial layout guide.

**Pull request merged**